### ISSUE TITLE
Не работает перевод в Yandex Disk

### ISSUE THREAD

**Issue by itelmenko** [ilyhalight/voice-over-translation#513]
### Operating system / Операционная система

Ubuntu 22

### Browser / Браузер

Firefox

### Userscript manager / Скриптовый движок

Tampermonkey

### Script version / Версия скрипта

1.5.0.5

### Describe the bug / Опишите ошибку

Появляется кнопка для перевода, но при нажатии получаю ошибку Failed to request video translation

Пример https://disk.yandex.ru/i/ZsepecBGpLxc4w

Доп. настройки в плагине делал.

Но и в Яндекс Браузере тоже не переводятся видео с Яндекс Диска.
Может быть поэтому и тут не работает?

**Comment by SashaXser** [comment]
> ### Operating system / Операционная система
> 
> Ubuntu 22
> ### Browser / Браузер
> 
> Firefox
> ### Userscript manager / Скриптовый движок
> 
> Tampermonkey
> ### Script version / Версия скрипта
> 
> 1.5.0.5
> ### Describe the bug / Опишите ошибку
> 
> Появляется кнопка для перевода, но при нажатии получаю ошибку Failed to request video translation
> 
> Пример https://disk.yandex.ru/i/ZsepecBGpLxc4w
> 
> Доп. настройки в плагине делал.
> 
> Но и в Яндекс Браузере тоже не переводятся видео с Яндекс Диска. Может быть поэтому и тут не работает?
> ### Additional information / Дополнительная информация
> 
> _No response_

    Для работы необходимо изменить опцию "Добавить TM в политику безопасности CSP" на "Да" (Tampermonkey -> Настройки -> Режим конфигурации: "Опытный" -> Добавить TM в политику безопасности CSP: "Да")
    Работает только с публичными ссылками

**Comment by itelmenko** [comment]
Как я писал выше, доп. настройки в плагине делал. Это НЕ помогает. А вы пробовали?
К тому же в самом Яндекс Браузере не переводятся видео с Яндекс Диска. Что-то мне подсказывает, что через плагин не будет работать тем более.

**Comment by Buumcode** [comment]
> К тому же в самом Яндекс Браузере не переводятся видео с Яндекс Диска. Что-то мне подсказывает, что через плагин не будет работать тем более.

В Windows 7 всё спокойно переводится. Только сегодня проверял. Видео должно быть общедоступным. Ваше видео у меня переводится. Браузер Опера.

**Comment by itelmenko** [comment]
Странно. А в Яндекс Браузере тоже переводится? Или только через плагин?
От чего может зависеть тогда?

**Comment by SashaXser** [comment]
В Firefox срабатывает CSP, в хромоподобных похоже всё работает нормально.

**Comment by itelmenko** [comment]
Хм... Заработало сейчас в Firefox. Не понятно с чем связано. Хорошо, извиняюсь за ложную тревогу. Закрываю issue

**Comment by itelmenko** [comment]
Интересно, что голос озвучки другой при переводе на Яндекс Диске. Не такой как при переводе Ютубовских роликов.

**Comment by SashaXser** [comment]
> Интересно, что голос озвучки другой при переводе на Яндекс Диске. Не такой как при переводе Ютубовских роликов.

Это из-за параметра "Авто", у него там своя схема подбора голосов.

**Comment by Buumcode** [comment]
> Странно. А в Яндекс Браузере тоже переводится? Или только через плагин?

В Яндекс-Браузере ничего не переводится – эта фича заблокирована Яндексом. А то сейчас миллиард чудиков со всего света начнут грузить видосы на диск и переводить – от чего лопнет и переводчик и сам диск (от перегрузки).
Данное расширение – это хак, исполняемый помимо воли Яндекса.

**Comment by itelmenko** [comment]
Чудики могут и на Ютуб грузить также. Какая разница то?
А как это так против воли Яндекса? Хоте ли бы, наверное, бы заблокировали

**Comment by Buumcode** [comment]
> Чудики могут и на Ютуб грузить также. Какая разница то?

Всё-таки разница есть. Любую гадость на ютуб в общий доступ не загрузишь. То есть условие "ютуб" отсеивает 90% попыток.
А против воли – да, хак дело такое. Он именно против воли. Хотели бы разрешить использовать на диске – разрешили бы.

**Comment by itelmenko** [comment]
Доступ в Ютуб можно сделать по ссылке (то есть не делать видео публичным). Можно загрузить все, что угодно.

**Comment by Buumcode** [comment]
> Доступ в Ютуб можно сделать по ссылке (то есть не делать видео публичным). Можно загрузить все, что угодно.

Перевод Яндекса работает только в ПУБЛИЧНОМ видео. С доступом по ссылке не работает.

**Comment by itelmenko** [comment]
Под публичным имеется в виду что ссылка доступна не только вам.
Просто загружаете на Ютуб и делаете доступным НЕ всем, а по ссылке.
И будет переводить в Яндекс Браузере.
Я пробовал

**Comment by Buumcode** [comment]
Да, вы правы. Доступ по ссылке – это же публичность, пусть и ограниченная. Сейчас попробовал – работает. 
Получается, Яндекс просто не хочет, чтоб грузили видео на его сервера. Дескать, пусть Ютуб отдувается )

**Comment by itelmenko** [comment]
Плагин поддерживает море сервисов. В то время как сам Яндекс Браузер - штук 5 сервисов переводи (https://yandex.ru/project/browser/streams/technology).

Я больше верю в то, что Яндекс пока не стремится нарастить объем запросов к переводчику, но не борется активно с текущим объемом и его динамикой.

**Comment by SergeyLuashkov** [comment]
У меня та же проблема, что и у автора, возможно я загружаю не правильное видео, т.е. нужны ли субтитры? и если нужны то нужно их встраивать в видео? или должно быть отдельным файлом? Настройки "Добавить TM в политику безопасности CSP: "Да"" я сделал

**Comment by itelmenko** [comment]
Субтитры не нужны. Яндекс распознает именно голос.

> Настройки "Добавить TM в политику безопасности CSP: "Да"" я сделал

Возможно стоит проверить сохранилась ли эта настройка. Не сбросилась ли

**Comment by Buumcode** [comment]
> У меня та же проблема, что и у автора, возможно я загружаю не правильное видео, т.е. нужны ли субтитры? и если нужны то нужно их встраивать в видео? или должно быть отдельным файлом? Настройки "Добавить TM в политику безопасности CSP: "Да"" я сделал

Вы сообщайте версию скрипта и ваш браузер.
-------- Видео надо делать публичным и открывать в отдельной вкладке: --------
Яндекс-Диск -- Только публичные ссылки. То есть вы заливаете файл на свой Яндекс-диск и делаете его доступным для других. Нажимаете "Скопировать ссылку", потом сами переходите по ней и смотрите с переводом. Яндекс-Диск отдаёт файл mp3 с дорожкой перевода, но поддержки субтитров на Яндекс-диске нет.

**Comment by andrei-zhdanov** [comment]
тоже не работает

**Comment by itelmenko** [comment]
> тоже не работает

Несколько постов выше - возможные пути решения.

**Comment by andrei-zhdanov** [comment]
> Несколько постов выше - возможные пути решения.

к сожалению все препробовал. выдает одну и ту же ошибку. "не удалось запросить перевод"

**Comment by Buumcode** [comment]
> к сожалению все препробовал. выдает одну и ту же ошибку. "не удалось запросить перевод"

Только что проверил – всё прекрасно работает. Вы, может быть, с какого-то неподдерживаемого языка пробуете переводить? Вот, попробуйте получить перевод для этого видео:
https://disk.yandex.ru/i/UL-cgN8NlWTSsA
-------- Видео надо делать публичным и открывать в отдельной вкладке: --------
Заливаете файл на свой Яндекс-диск и делаете его доступным для других.
Нажимаете "Скопировать ссылку", потом сами переходите по ней и смотрите с переводом.

**Comment by itelmenko** [comment]
Говорят, что в новой версии Tamper Monkey (5.2) нет варианта Yes для CSP. Есть только Auto, с которым перевод не работает. Выход пока - ставить Legacy версию.
